Formula/synergy-core.rb

```diff
@@ -0,0 +1,376 @@
+      server_log = [log_trim.call(io_server.gets)]
+      permission_denied = (server_log[0] == log_permission_error)
+      if permission_denied
+        # We receiced the message indicating a lack of 'Accessibility' permission.
+        # Verify expected exit status. It should be 1.
+        _, status = Process.wait2(io_server.pid)
+        assert_equal status.exitstatus, 1
+
+        # We want the user to know about this condition so that they know what
+        # they have to do if they want to run more detailed testing.
+        ohai "The current process does not have the macOS 'Accessibility' permission."
+      elsif server_log[0] == log_no_display_error
+        # In this condition, Synergy locks up and declines to close.
+        kill_and_wait_for io_server
+
+        ohai "Synergy believes that your system is headless."
+        ohai "Without a display, Synergy's server and client won't run."
+
+        # No further tests can be run.
+        return true
+      else
+        # The potentially-anomalous keyboard and mouse behavior can be surprising to
+        # users. To avoid surprise, we offer this warning.
+        opoo "Your mouse and keyboard may behave anomalously for the next few seconds"
+        opoo "while running these synergy-core tests. Your mouse cursor may disappear."
+      end
+      puts
+
+      # Spawn Synergy client.
+      IO.popen [
+        opt_bin/"synergyc",
+        "-f", # run in foreground
+        "-n", "test-client",
+        "127.0.0.1:#{port}", # must be final command line argument to synergyc
+        err: [:child, :out] # redirect child's stderr to its stdout
+      ] do |io_client|
+        # If the Synergy server has the 'Accessibility' permission, this test will
+        # cause the user's mouse and keyboard to exhibit unexpected behavior. To minimize
+        # this side effect, we will kill the client and server after a couple seconds.
+        #
+        # Also, regardless of whether we have the 'Accessibility' permission, the
+        # Synergy programs tend to stay open indefinitely and refrain from closing,
+        # even if they receive the TERM signal. We will use the KILL signal to make
+        # sure they close.
+        sleep 2
+        kill_and_wait_for io_client
+        unless permission_denied
+          sleep 1 # Give server time to recognize that client has disconnected.
+          kill_and_wait_for io_server
+        end
+
+        # Collect log messages from the server and client.
+        server_log.push(*io_server.readlines.map!(&log_trim)) unless permission_denied
+        server_set = server_log.to_set
+        client_log = io_client.readlines.map!(&log_trim)
+        client_set = client_log.to_set
+
+        # Print the logs so that a human can potentially spot issues not caught
+        # by the automated testing. Indent the log lines so that it is easier to
+        # spot the headers that separate the server output from the client output.
+        #
```

Comment:
There is a near 0% chance people will look at this instead of trying to reproduce the issues separately. Let's just remove it.